discord/automod.py

```diff
@@ -89,16 +91,43 @@ def __init__(self, *, channel_id: Optional[int] = ...) -> None:
         ...
 
     @overload
-    def __init__(self, *, duration: Optional[datetime.timedelta] = ...) -> None:
+    def __init__(
+        self, *, type: Literal[AutoModRuleActionType.send_alert_message] = ..., channel_id: Optional[int] = ...
+    ) -> None:
+        ...
+
+    @overload
+    def __init__(self, *, duration: datetime.timedelta = ...) -> None:
+        ...
+
+    @overload
+    def __init__(
+        self, *, type: Literal[AutoModRuleActionType.timeout] = ..., duration: Optional[datetime.timedelta] = ...
+    ) -> None:
         ...
 
     @overload
     def __init__(self, *, custom_message: Optional[str] = ...) -> None:
         ...
 
+    @overload
+    def __init__(self, *, type: Literal[AutoModRuleActionType.block_message] = ...) -> None:
+        ...
+
+    @overload
+    def __init__(
+        self, *, type: Literal[AutoModRuleActionType.block_message] = ..., custom_message: Optional[str] = ...
+    ) -> None:
+        ...
+
+    @overload
+    def __init__(self, *, type: Literal[AutoModRuleActionType.block_member_interactions] = ...) -> None:
+        ...
+
     def __init__(
```

Comment:
I would change the first error from
```py
raise ValueError(f'{attribute} can only be passed if type is {action_type.name}.')
```
to
```py
raise ValueError(f'{attribute} can only be passed if type is {action_type.name} or None.')
```
Otherwise I think this is good, thanks!

Reply:
One thing I noticed in 
```py
if type is None and sum(v is None for v in (channel_id, duration, custom_message)) < 2:
   ^^^^^^^^^^^^
```
`type` is irrelevant since it also applies when `type` is not `None`.